### ISSUE TITLE
Revisit "Add :after_commit callbacks to events"

### ISSUE THREAD

**Issue by swrobel** [aasm/aasm#846]
There was a pull request #367 that improves the behavior of `after_commit` callbacks with ActiveRecord. Supposedly the old maintainer merged it, but it doesn't seem to exist in the current codebase. There is a good description & discussion on that PR, but it has probably been missed because it's been closed for some time. cc: @axsuul

**Comment by marythought** [comment]
Yes please! Using an old forked version implementing this change and would love to be able to update the gem.
